tasks/ensure_k3s_auto_deploy.yml

```
@@ -34,7 +34,7 @@
 - name: Ensure auto-deploying manifests are downloaded to the primary controller
   ansible.builtin.get_url:
     url: "{{ item.url }}"
-    dest: "{{ k3s_server_manifests_dir }}/{{ item.filename }}"
+    dest: "{{ k3s_server_manifests_dir }}/{{ item.filename | default(item.url | basename }}"
```

```suggestion
    dest: "{{ k3s_server_manifests_dir }}/{{ item.filename | default(item.url | basename) }}"
```

I think it's missing a bracket